tools/shared/src/main/scala/org/scalajs/core/tools/sourcemap/SourceMapWriter.scala

```diff
@@ -142,13 +168,14 @@ class SourceMapWriter(
     lastColumnInGenerated = pendingColumnInGenerated
 
     // If the position is NoPosition, stop here
-    if (!pendingPos.isDefined)
+    val pendingPos1 = pendingPos
+    if (!pendingPos1.isDefined)
       return
 
     // Extract relevant properties of pendingPos
-    val source = pendingPos.source
-    val line = pendingPos.line
-    val column = pendingPos.column
+    val source = pendingPos1.source
+    val line = pendingPos1.line
+    val column = pendingPos1.column
 
     // Source index field
     if (source == lastSource) { // highly likely
```

Comment:
Maybe try `source eq lastSource` here? Positions coming from different `.sjsir` files are unlikely to have equal `source`s (although it's possible, because of inner classes), and all positions coming from the same `.sjsir` will be `eq`, not just `==`. So the speed improvement due to using `eq` might outweigh the increased "cache misses".

This absolutely needs profiling to decide which one is best.